examples/pytorch/GatedGCN/main.py

```diff
@@ -0,0 +1,129 @@
+import argparse
+import time
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from model import GatedGCN
+from ogb.graphproppred import DglGraphPropPredDataset, Evaluator, collate_dgl
+from torch.utils.data import DataLoader
+
+
+def train(model, device, data_loader, opt, loss_fn):
+    model.train()
+
+    train_loss = []
+    for g, labels in data_loader:
+        g = g.to(device)
+        labels = labels.to(torch.float32).to(device)
+        logits = model(g, g.edata["feat"], g.ndata["feat"])
+        loss = loss_fn(logits, labels)
+        train_loss.append(loss.item())
+
+        opt.zero_grad()
+        loss.backward()
+        opt.step()
+
+    return sum(train_loss) / len(train_loss)
+
+
+@torch.no_grad()
+def evaluate(model, device, data_loader, evaluator):
+    model.eval()
+    y_true, y_pred = [], []
+
+    for g, labels in data_loader:
+        g = g.to(device)
+        logits = model(g, g.edata["feat"], g.ndata["feat"])
+        y_true.append(labels.detach().cpu())
+        y_pred.append(logits.detach().cpu())
+
+    y_true = torch.cat(y_true, dim=0).numpy()
+    y_pred = torch.cat(y_pred, dim=0).numpy()
+
+    return evaluator.eval({"y_true": y_true, "y_pred": y_pred})["rocauc"]
+
+
+def main():
+    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+    # load ogb dataset & evaluator
+    dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
+    evaluator = Evaluator(name="ogbg-molhiv")
+
+    n_classes = dataset.num_tasks
+
+    split_idx = dataset.get_idx_split()
+    train_loader = DataLoader(
+        dataset[split_idx["train"]],
+        batch_size=args.batch_size,
+        shuffle=True,
+        collate_fn=collate_dgl,
+    )
+    valid_loader = DataLoader(
+        dataset[split_idx["valid"]],
+        batch_size=args.batch_size,
+        shuffle=False,
+        collate_fn=collate_dgl,
+    )
+
+    test_loader = DataLoader(
+        dataset[split_idx["test"]],
+        batch_size=args.batch_size,
+        shuffle=False,
+        collate_fn=collate_dgl,
+    )
+
+    # load model
+    model = GatedGCN(
+        hid_dim=args.hid_dim,
+        out_dim=n_classes,
+        num_layers=args.num_layers,
+        dropout=args.dropout,
+    ).to(device)
+
+    print(model)
+
+    opt = optim.Adam(model.parameters(), lr=args.lr)
+    loss_fn = nn.BCEWithLogitsLoss()
+
+    print("---------- Training ----------")
+    for i in range(args.epochs):
+        t1 = time.time()
+        train_loss = train(model, device, train_loader, opt, loss_fn)
+        elapsed_time = time.time() - t1
+
+        train_auc = evaluate(model, device, train_loader, evaluator)
+        valid_auc = evaluate(model, device, valid_loader, evaluator)
+        test_auc = evaluate(model, device, test_loader, evaluator)
+
+        print(
+            f"Epoch {i} | Elapsed Time {elapsed_time:.2f} | Train Loss: {train_loss:.4f} | Train Auc: {train_auc:.4f} | Valid Auc: {valid_auc:.4f} | Test Auc: {test_auc:.4f}"
```

Comment:
Follow the pattern here, so it is easier to write regression test for this example:
https://github.com/dmlc/dgl/blob/f41934df87c5c38ee0286ce6807fb8eedcf78ac1/examples/sparse/sign.py#L89